### PR TITLE
EXL-375 display menu items based on profile

### PR DESCRIPTION
**Pull request by zsso**
- Modified visible menu items based on roles

- Removed rights from super-admin to hide some menu items

- Fixed tickets view to display popup where Tehnician alocat is empty

- Fixed ticket permission for subteh. to open tickets where he is teh. park

### CODE REVIEW

**Comment by zsso** [comment]
This is not ready yet.

File: plugin/inc/profile.class.php [hunk 1]
```diff
@@ -76,9 +75,9 @@ function right_array($right, $profiles)
             'plugin_iservice_view_intorders' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
             'plugin_iservice_view_extorders' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
             'plugin_iservice_view_reminders' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
-            'plugin_iservice_view_cartridges' => right_array(ALLSTANDARDRIGHT, $all_profiles),
+            'plugin_iservice_view_cartridges' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
             'plugin_iservice_view_emaintenance' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
-            'plugin_iservice_view_global_readcounter' => right_array(ALLSTANDARDRIGHT, array_merge($level15_profiles, $elevated_profiles)),
+            'plugin_iservice_view_global_readcounter' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
```
Comment (hupupu) [inline plugin/inc/profile.class.php:80]: The global_readcounter must be visible for the `superclient` and `subtehnician` also, isn't it?

File: plugin/inc/ticket.class.php [hunk 2]
```diff
@@ -45,6 +45,17 @@ class PluginIserviceTicket extends Ticket
...
 
+    public function canViewItem(): bool
+    {
+        return parent::canViewItem() || $this->isUserTechPark();
+    }
+
+    public function isUserTechPark(): bool
+    {
+        $printer = $this->getFirstPrinter();
+        return !empty($printer->fields['users_id_tech']) && $printer->fields['users_id_tech'] === Session::getLoginUserID();
```
Comment (hupupu) [inline plugin/inc/ticket.class.php:56]:
```suggestion
        return ($printer->fields['users_id_tech'] ?? '')=== Session::getLoginUserID();
```

File: plugin/inc/views/tickets.class.php [hunk 3]
```diff
@@ -322,7 +322,7 @@ protected function getSettings(): array
                             , u.id tech_park_id
                             , CONCAT(IFNULL(CONCAT(u.realname, ' '),''), IFNULL(u.firstname, '')) tech_park_name
                             , a.id tech_assign_id
-                            , CONCAT(IFNULL(CONCAT(a.realname, ' '),''), IFNULL(a.firstname, '')) tech_assign_name
+                            , CONCAT(IFNULL(CONCAT(a.realname, ' '),''), IFNULL(a.firstname, ''), '-') tech_assign_name
```
Comment (hupupu) [inline plugin/inc/views/tickets.class.php:325]: Why do we need a `-` to the end of the name?

**Review by hupupu — CHANGES_REQUESTED** [review]

File: plugin/inc/profile.class.php (hunk 1 above)
Comment (zsso) [reply plugin/inc/profile.class.php:80]: Zoli said that only solicitare, tickets, aparate should be visible for them.

Comment (hupupu) [reply plugin/inc/profile.class.php:80]: Let's double check with him

**Review by hupupu — DISMISSED** [review]

File: plugin/inc/profile.class.php [hunk 4]
```diff
@@ -77,7 +77,7 @@ function right_array($right, $profiles)
             'plugin_iservice_view_reminders' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
             'plugin_iservice_view_cartridges' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
             'plugin_iservice_view_emaintenance' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
-            'plugin_iservice_view_global_readcounter' => right_array(ALLSTANDARDRIGHT, $elevated_profiles),
+            'plugin_iservice_view_global_readcounter' => right_array(ALLSTANDARDRIGHT, array_merge($level15_profiles, $elevated_profiles)),
```
Comment (hupupu) [inline plugin/inc/profile.class.php:80]:
```suggestion
            'plugin_iservice_view_global_readcounter' => right_array(ALLSTANDARDRIGHT, $all_profiles)),
```

**Review by hupupu — APPROVED** [review]

**Review by hupupu — APPROVED** [review]

**Pull request merged**